maggma/stores.py

```diff
@@ -561,13 +582,16 @@ def query(self, properties=None, criteria=None, **kwargs):
                 against key-value pairs
             **kwargs (kwargs): further kwargs to Collection.find
         """
+        if isinstance(criteria, dict):
+            self.transform_criteria(criteria)
         for f in self.collection.find(filter=criteria, **kwargs):
             data = f.read()
+            metadata = f.metadata
             try:
-                json_dict = json.loads(data)
-                yield json_dict
+                data = json.loads(data)
             except:
-                yield data
+                pass
+            yield dict(data=data, metadata=metadata)
```

Comment:
It shouldn't actually be necessary to yield back the metadata, since the keys put into the gridfs  metadata will either already be in the doc or be an accessory for GridFSStore such as 'compression'